src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6322,12 +6344,167 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
+                LeaveRegionsUpTo(resultRegion);
+                AppendNewBlock(new BasicBlockBuilder(BasicBlockKind.Block), linkToPrevious: true);
+            }
+            else
+            {
+                LeaveRegionsUpTo(resultRegion);
+            }
+
+            if (operation.HandlerCreationHasSuccessParameter)
+            {
+                RemovePlaceholder(outParameterPlaceholder!);
+            }
+
+            var appendCalls = collectAppendCalls(operation);
+
+            int appendCallsLength = appendCalls.Length;
+            for (var i = 0; i < appendCallsLength; i++)
+            {
+                var appendCall = appendCalls[i];
+                if (operation.HandlerAppendCallsReturnBool)
+                {
+                    Debug.Assert(resultBlock != null);
+
+                    if (i == appendCallsLength - 1)
+                    {
+                        // No matter the result, we're going to the result block next. So just visit the statement, and if the current block can be
+                        // combined with the result block, the compaction machinery will take care of it
+                        VisitStatement(appendCall.AppendCall);
+                        AppendNewBlock(resultBlock, linkToPrevious: true);
+                    }
+                    else
+                    {
+                        // if (!appendCall()) goto result else goto next block
+                        ConditionalBranch(VisitRequired(appendCall.AppendCall), jumpIfTrue: false, resultBlock);
+                        AppendNewBlock(new BasicBlockBuilder(BasicBlockKind.Block), linkToPrevious: true);
+                    }
+                }
+                else
+                {
+                    VisitStatement(appendCall.AppendCall);
+
+                    if (i == appendCallsLength - 1 && operation.HandlerCreationHasSuccessParameter)
+                    {
+                        Debug.Assert(resultBlock != null);
+                        AppendNewBlock(resultBlock, linkToPrevious: true);
+                    }
+                }
+            }
+
+            return new FlowCaptureReferenceOperation(handlerCaptureId, operation.Syntax, operation.Type, operation.GetConstantValue());
+
+            static ImmutableArray<IInterpolatedStringAppendOperation> collectAppendCalls(IInterpolatedStringHandlerCreationOperation creation)
+            {
+                var appendCalls = ArrayBuilder<IInterpolatedStringAppendOperation>.GetInstance();
+                if (creation.Content is IInterpolatedStringOperation interpolatedString)
+                {
+                    // Simple case
+                    appendStringCalls(interpolatedString, appendCalls);
+                    return appendCalls.ToImmutableAndFree();
+                }
+
+                var stack = ArrayBuilder<IInterpolatedStringAdditionOperation>.GetInstance();
+                pushLeftNodes((IInterpolatedStringAdditionOperation)creation.Content, stack);
```

Comment:
Consider asserting that this is an `IInterpolatedStringAdditionOperation` beforehand

Reply:
I don't think that's particularly necessary. The code will just fail here.